### DISCUSSION TITLE
Full House from scratch in IfcOpenShell python

### DISCUSSION THREAD

**Discussion by Celekohr** [IfcOpenShell/IfcOpenShell#2775] [Q&A]
<img width="834" alt="Bildschirm­foto 2023-02-15 um 14 37 40" src="https://user-images.githubusercontent.com/116890819/219042534-4b851c66-eaec-4150-9948-f811abc946d7.png">
There is this house I found in ifcopenshell and I want to ask if there is the source code for python to build this house from scratch just using code.

**Comment by aothms** [comment]
This is "FZK hause". It's not programmed. https://www.ifcwiki.org/index.php?title=KIT_IFC_Examples

We do have IfcOpenHouse https://raw.githubusercontent.com/aothms/IfcOpenHouse/master/IfcOpenHouse_IFC2x3.ifc

Programmed in C++ https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.7.0/src/examples/IfcOpenHouse.cpp

These days though with the Python high level API it would be a much easier task.

**Reply by Celekohr** [reply]
is there the source code for the IfcOpenHouse the guide on the page seamed to be not complete

**Comment by Celekohr** [comment]
Thx for the fast answer is there a in python api programmed house source code out there.

**Reply by Moult** [reply]
This code sample might help you get started https://blenderbim.org/docs-python/ifcopenshell-python/code_examples.html#create-a-simple-model-from-scratch

**Comment by Celekohr** [comment]
And do you think it is the best and fastest way to use the ifcopenshell.api to create a house from scratch 

Great 

Clemens

**Reply by aothms** [reply]
I'm quite sure, there doesn't exist anything like it :) https://blenderbim.org/docs-python/autoapi/ifcopenshell/api/geometry/index.html

I'm sure the community can help out in the process.

**Reply by CamilleZ99** [reply]
> I'm quite sure, there doesn't exist anything like it :) https://blenderbim.org/docs-python/autoapi/ifcopenshell/api/geometry/index.html
> 
> I'm sure the community can help out in the process.

is there any source code of building a full house in ifcOpenshell? looks like what you provide is either ifc or sorce code of building a single wall. Is there any sample code in ifcOpenshell? thanks a lot.
